platforms/core-configuration/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/PluginDependenciesSpecExtensions.kt

```diff
@@ -25,11 +25,27 @@ import org.gradle.plugin.use.PluginDependencySpec
  *
  * Visit the [Build Scan Plugin User Manual](https://docs.gradle.com/build-scan-plugin/) for additional information.
  *
- * By default, the applied plugin version will be the same as the one used by the `--scan` command line option.
+ * By default, the applied plugin version will be "3.16.2", which is the last version of the Gradle Enterprise plugin.
+ * To apply later versions, use `develocity` plugin instead.
  *
  * You can also use e.g. `` `gradle-enterprise` version "3.0" `` to request a different version.
  *
  * @since 6.0
  */
 val PluginDependenciesSpec.`gradle-enterprise`: PluginDependencySpec
```

Comment:
💬 This seems to be the reason behind [this failure](https://ge.gradle.org/s/veaxzcxf6s2f6/console-log?page=13#L12094-L12096). What are we supposed to do with this?

Reply:
It seems like this change can only be made on `master`. Is it right?

Reply:
No, this only means you need to accept the binary change.

Reply:
@wolfs , How can I do that to unblock the sanity check?

Reply:
Do we also need a DeprecationLogger nag here?

Reply:
Check out the Binary compatibility report. It has a button `Accept Changes for all Errors` and instructions how to fix it. 
<img width="1038" alt="image" src="https://github.com/gradle/gradle/assets/423186/93d697b7-d228-41c3-aac0-7d70af95ab24">

Reply:
Actually, `@since` and `@Incubating` are missing from the new method. 
We should definitely add `@since`.
We can now either add `@Incubating`, or accept the binary change and deprecate the old method. If we add `@Incubating`, then we should wait at least one Gradle release until we remove the `@Incubating` and deprecate the old method.

<img width="1356" alt="image" src="https://github.com/gradle/gradle/assets/423186/bd0e5e6d-604a-4c6d-856a-b2f9043ea638">

Reply:
I am currently on the fence what to do for 8.8. Let's start out with having the new `develocity` method as incubating and the old one non-deprecated. And then let's do the deprecation for 8.9.

Reply:
I pushed a commit to do just that.

Reply:
How does this interact with static plugin block parsing? Do we have special handling for this method? Do we need to add deprecation logging to the interpreter?

Reply:
@wolfs , may I ask you to answer Justin's questions? I am more than happy to make any necessary changes, but I could use some guidance.

Reply:
> How does this interact with static plugin block parsing?

What is the static plugin block parsing? Is this something from Kotlin DSL? If yes, then for Kotlin DSL we expose the methods here to the plugins block. At least that is my understanding.

Reply:
@bamboo would know better but afaik in the kotlin dsl, for simple plugins blocks, we parse and interpret the block without running it through the kotlin compiler/bytecode execution.